ProjFS.Mac/PrjFSKext/public/Message.h

```diff
@@ -47,17 +47,38 @@ struct MessageHeader
     int32_t             pid;
     char                procname[MAXCOMLEN + 1];
 
-    // Size of the flexible-length, nul-terminated path following the message body, including the nul character.
-    uint16_t            pathSizeBytes;
+    // Sizes of the flexible-length, nul-terminated paths following the message body.
+    // Sizes include the nul characters but can be 0 to indicate total absence.
+    union
+    {
+        struct
+        {
+            // Current/destination path
+            uint16_t pathSizeBytes;
+            // Original location (rename, hard link)
+            uint16_t sourcePathSizeBytes;
+        };
+        uint16_t     stringSizesBytes[2];
+    };
 };
 
 // Description of a decomposed, in-memory message header plus variable length string field
 struct Message
 {
     const MessageHeader* messageHeader;
-    const char*    path;
+    union
+    {
+        struct
+        {
+            const char* path;
+            const char* sourcePath;
+        };
+        const char*     strings[2];
```

Comment:
Whats the advantch here over:
const char* path;
const char* sourcePath;

Reply:
See [my comment here](https://github.com/Microsoft/VFSForGit/pull/985#discussion_r275348890) on how this turned out.